### PR TITLE
feat(fsdp): add tensor parallel for FSDP trainer

### LINKED ISSUE #278: [Feature]:  Add tensor parallel for embodiment model rollout
### Feature description

torch provides `parallelize_module`([docs](https://docs.pytorch.org/docs/2.6/distributed.tensor.parallel.html)) to support tensor model parallel, maybe we could use this to implement tensor parallel in embodiment model to lower down VRAM requirements.

### Additional context

_No response_

### Before submitting a new issue...

- [x] Have you checked relevant issues, FAQs (https://rlinf.readthedocs.io/en/latest/rst_source/faq.html), or asked the chatbot at the top right corner of the [documentation page](https://rlinf.readthedocs.io).

### PR DESCRIPTION
**Pull request by anHappyDog**
### Description
Currently for math reasoning, FSDP can not train models with long text sequence(like 27k or longer) due to generated activations and logits. Use torch's original tensor parallel to lower down VRAM requirements.

### How has this been tested?
- [ ] corresponding documents
- [ ] ci test
- [ ] tested in single machine 

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Documentation update (Document-only update)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by guozhen1997** [comment]
Closing this PR due to prolonged inactivity. Feel free to reopen it if you’d like to continue working on it.
